src/graphql_resolvers.rs

```diff
@@ -16,39 +30,84 @@ struct Transaction {
-impl juniper::Context for Context {}
-impl Context {
-    pub fn new() -> Self {
-        Context {}
-    }
+pub struct Context {
+    pub pool: database::DbPool,
+    pub jwt_secret: String,
 }
+impl juniper::Context for Context {}
 
 pub struct Query;
 
 #[graphql_object(context = Context)]
 impl Query {
-    fn apiVersion() -> &'static str {
+    async fn apiVersion() -> &'static str {
         "1.0"
     }
 
-    fn transactions(context: &Context, user_uid: String) -> FieldResult<Vec<Transaction>> {
+    async fn transactions(context: &Context, user_uid: String) -> FieldResult<Vec<Transaction>> {
         let parsed_user_uid = Uuid::parse_str(&user_uid)?;
-        let transactions = services::transaction::list_user_transactions(parsed_user_uid)
-            .iter()
-            .map(|t| Transaction {
-                id: t.id,
-                related_user: t.related_user,
-                entry_date: t.entry_date,
-                entry_account_code: t.clone().entry_account_code,
-                exit_account_code: t.clone().exit_account_code,
-                amount: t.amount,
-                description: t.clone().description,
-            })
-            .collect();
+        let transactions =
+            services::transaction::list_user_transactions(&context.pool, parsed_user_uid)
+                .iter()
+                .map(|t| Transaction {
+                    id: t.id,
+                    related_user: t.related_user,
+                    entry_date: t.entry_date,
+                    entry_account_code: t.clone().entry_account_code,
+                    exit_account_code: t.clone().exit_account_code,
+                    amount: t.amount,
+                    description: t.clone().description,
+                })
+                .collect();
         Ok(transactions)
     }
+
+    async fn token(context: &Context, email: String, login_code: i32) -> FieldResult<String> {
+        let token = services::user::validate_and_generate_token(
+            &context.pool,
+            email,
+            login_code,
+            &context.jwt_secret,
+        )?;
+        Ok(token)
+    }
+}
+
+pub struct Mutations;
+
+#[graphql_object(context = Context)]
+impl Mutations {
+    async fn create_user(context: &Context, username: String, email: String) -> FieldResult<User> {
+        let user = models::user::NewUser {
+            username,
+            register_date: None,
+            email,
+            last_code_gen_request: None,
+            login_code: None,
+        };
+        let created_user = services::user::create_user(&context.pool, user)?;
+        Ok(User {
+            id: created_user.id,
+            username: created_user.username,
+            register_date: created_user.register_date,
+            email: created_user.email,
+            last_code_gen_request: created_user.last_code_gen_request,
+            login_code: created_user.login_code,
+            is_registered: created_user.is_registered,
+        })
+    }
```

Comment:
In the future, we will need to be more careful about adding users in the DB without any type of validation, like a captcha. Just saying, nothing to change now.